CRM/Civicase/Service/CaseCategoryFromUrl.php

```diff
@@ -232,9 +232,10 @@ private function getCaseCategoryForAjaxRequest($caseTypeCategoryParam) {
 
         if (strtolower($entityName) == 'case') {
           $this->isCaseEntity = TRUE;
-          if (isset($params[$caseTypeCategoryParam])) {
-            return $this->getCaseTypeCategoryNameFromOptions($params[$caseTypeCategoryParam]);
-          }
+        }
+
+        if (isset($params[$caseTypeCategoryParam])) {
+          return $this->getCaseTypeCategoryNameFromOptions($params[$caseTypeCategoryParam]);
```

Comment:
Not sure we should change this. The `'case_type_id.case_type_category'` is only useful for the case entity. The `Activity`, `Contact` entity for instance is not related to this parameter.

Reply:
Not applicable after refactoring.